### PR TITLE
Add parseDate option

### PR DESCRIPTION
**Pull request by dsl101**
Finally after only 2 years :). I needed this on another project, and eventually got round to making a PR.

### CODE REVIEW

File: lib/feedparser/index.js [hunk 1]
```diff
@@ -71,6 +71,7 @@ function FeedParser (options) {
   if (!('strict' in this.options)) this.options.strict = false;
   if (!('normalize' in this.options)) this.options.normalize = true;
   if (!('addmeta' in this.options)) this.options.addmeta = true;
+  if (!('parseDate' in this.options)) this.options.parseDate = true;
```
Comment (danmactough) [inline lib/feedparser/index.js]: We've already parsed the date string into a `Date` object. With this option enabled, we'll actually be stringifying the date. So, let's rename this option: maybe `stringifydate`. (If you have a different suggestion, that's ok, but let's also ensure the option names are consistent -- one word, all lowercase.) (I know, I know, there's `resume_saxerror` which has an underscore -- 🤷‍♂️ )

Also, we need to preserve the current behavior as the default, so if the option is not provided, it's value should be `false`.

File: lib/feedparser/index.js [hunk 2]
```diff
@@ -430,6 +431,7 @@ FeedParser.prototype.handleMeta = function handleMeta (node, type, options) {
 
   var meta = {}
     , normalize = !options || (options && options.normalize)
+    , parseDate = !options || (options && options.parseDate)
```
Comment (danmactough) [inline lib/feedparser/index.js]: `parseDate = !options || (options && options.parseDate)`

This logic will need to be flipped: `stringifydate` should be true only if `options && options.stringifydate` is true.

File: lib/feedparser/index.js [hunk 3]
```diff
@@ -777,6 +780,7 @@ FeedParser.prototype.handleItem = function handleItem (node, type, options){
 
   var item = {}
     , normalize = !options || (options && options.normalize)
+    , parseDate = !options || (options && options.parseDate)
```
Comment (danmactough) [inline lib/feedparser/index.js]: Same comment as above.

**Comment by dsl101** [comment]
Yes, that's fine. I was going by what you'd originally said [here](https://github.com/danmactough/node-feedparser/issues/180#issuecomment-350617547) :). Maybe I misunderstood what you meant? Anyway, I'm away for a few days now but will edit when I'm back.

My reason for doing this is all firebase related. When used in a callable firebase cloud function, the date objects in the response from feedparser are all returned to my client-side app as empty objects for some reason. I suspect maybe a bug in firebase, but to work round I have to convert all dates to strings before the function returns. ISO date will work for me, but to make it more flexible, it could be an option called `dateformat` which, if omitted, leaves it as a date object, but could have values like `toISOString`, 'toLocaleString`, etc. to match the native string formats that the Date object can convert to. What do you think?

**Comment by danmactough** [comment]
> Yes, that's fine. I was going by what you'd originally said here :). Maybe I misunderstood what you meant?

Nope. You understood fine. I guess I'm  ... mercurial.  😂

**Comment by dsl101** [comment]
This is superseded by the other PR. Closing this one.

**Pull request closed without merging**